### ISSUE TITLE
Add old and new SCDB data

### ISSUE THREAD

**Issue by mlissner** [freelawproject/courtlistener#535]
The 2015 term is now out, and a few months ago they released everything back to 1791. Both need to be added.

Some problems/tasks I've identified:
- [x] There's a setting in 10-public.py that lists the most recent case in SCDB. This needs to be updated.
- [x] In the import script, there are two issues:
  - [x] [This section of code](https://github.com/freelawproject/courtlistener/blob/master/cl/corpus_importer/management/commands/update_scdb.py#L107-L134) should be rewritten into a single query.
  - [x] [This section of code](https://github.com/freelawproject/courtlistener/blob/master/cl/corpus_importer/management/commands/update_scdb.py#L107-L134) is shoddy because it could create duplicate identical values in the DB.
- [x] This introduces SCDB dates prior to 1900, raising the specter of the pre-1900's Python bug. We'll need to make sure we won't run into that.

We can use this ticket to keep track of other tasks that are involved since we'll be doing this every year or so.

**Comment by mlissner** [comment]
`strftime` is verified to be OK. I combed the code for it and only found references to how annoying it is.

**Comment by mlissner** [comment]
I've condensed the part of the code that should have been a single query.

**Comment by mlissner** [comment]
The 2016 SCDB is added as is the Legacy data set. I need to go back through the log and investigate the situation, but this has been a huge improvement. In the end, there were about 200 items that I did by hand, resolving duplicates that no computer could do correctly. 

It's worth it though: If we added 20k items from SCDB, each with 2-3 citations, we can figure that the system now has about 50-60k more citations than it used to. Pretty great.

**Comment by mlissner** [comment]
I went through and investigated about 2000 warnings that the import script generated, and I put notes in the log file that I have locally. The next step is to re-create the Solr database and to then re-do the citation finder. The former is started and I'll get a push notification when it finishes. At that point I will kick off the citation finder. In any case, I'm calling this done.

**Comment by mlissner** [comment]
Blogged here: https://free.law/2016/09/06/courtlisteners-scotus-data-gets-even-better-with-legacy-data-from-the-supreme-court-database/

**Comment by mlissner** [comment]
Reopening. It's high time we get this imported for 2016.

Process:
 - [ ] Identify any corrections in the old data by diffing the 2015 and 2016 datasets.
 - [ ] Import the latest data
 - [ ] Update the settings with the latest date.
 - [ ] Re-run the citation finder for any case that might have cited to these citations
 - [ ] Update solr with the latest citations

**Comment by mlissner** [comment]
This is done again. A couple of notes:

1. I rewrote parts of the `update_scdb` script that were wonky, so it should be better now, including eliminating at least one minor bug.

1. I attempted to diff the old CSV with the new, but it was too much of a mess. The rows are very long, and figuring out the difference was a pain. Turns out this wasn't necessary because the `update_scdb` script does this anyway.

1. After importing I reviewed the logs for any warnings and manually resolved any changes. Nothing much of significance was there. 

1. Once that was done, I generated a list of all the Opinion objects whose docket, cluster, or opinion were modified after today. I passed that into the `cl_find_citations` command using the `--doc_id` parameter and the `--index concurrently` flags.
